pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -228,14 +228,22 @@ message PodSandboxConfig {
     repeated PortMapping port_mappings = 5;
     // Key-value pairs that may be used to scope and select individual resources.
     map<string, string> labels = 6;
-    // Annotations is an unstructured key value map that may be set by external
-    // tools to store and retrieve arbitrary metadata. There are a few features are
-    // driven by annotations, Runtimes could support them optionally:
+    // Unstructured key value map that may be set by external tools to store
+    // and retrieve arbitrary metadata. Annotations MUST NOT be altered or in
+    // any way processed by the runtime.
+    //
+    // In order to preserve a well-defined interface between the kubelet and
+    // the container runtime, annotations SHOULD NOT influence runtime behaviour.
```

Comment:
tried to capture the spirit of that with "Runtime authors should similarly consider proposing new fields for any new features instead of using annotations."

also, it's very intentional that this is a SHOULD and not a MUST